### ISSUE TITLE
Update youtube-dl the proper way

### ISSUE THREAD

**Issue by mbnoimi** [axcore/tartube#41]
Hi,

Whenever I tried to update `youtube-dl` I get this error message:
```
Starting update operation, installing/updating youtube-dl
Collecting youtube-dl
  Using cached https://files.pythonhosted.org/packages/ef/f8/3424069c6d1e285fda98a83c48470578d2df4047e6e77e1b16fc28cf95d3/youtube_dl-2020.1.15-py2.py3-none-any.whl
Installing collected packages: youtube-dl
  Found existing installation: youtube-dl 2020.1.1
    Uninstalling youtube-dl-2020.1.1:
ERROR: Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/usr/bin/youtube-dl'
Consider using the `--user` option or check the permissions.
Child process exited with non-zero code: 1
Update operation finished
```
I use the same configurations mentioned in README file:
https://raw.githubusercontent.com/axcore/tartube/master/screenshots/example2.png

I tried to use the following configurations but I got another error message:
![image](https://user-images.githubusercontent.com/284547/72425257-7d011100-3798-11ea-83f4-2c004656d2b2.png)
```
Starting update operation, installing/updating youtube-dl
It looks like you installed youtube-dl with a package manager, pip, setup.py or a tarball. Please use that to update.
Update operation finished
```

Of course I can update it manually but I wonder why this feature is available while Tartube unable to update `youtube-dl` by itself?!

**Comment by axcore** [comment]
On the same computer, can you update youtube-dl using youtube-dl-gui?

**Comment by mbnoimi** [comment]
> On the same computer, can you update youtube-dl using youtube-dl-gui?

I don't know, I removed it for good hehehheehe

**Comment by axcore** [comment]
>I wonder why this feature is available while Tartube unable to update youtube-dl by itself?!
!

I am also using Linux Mint 19.2, and it is working for me:

[tartube1](https://user-images.githubusercontent.com/37951926/72998847-64c36e80-3df6-11ea-802a-ee03dcbf5e25.png)

Using these settings:

![tartube2](https://user-images.githubusercontent.com/37951926/72998887-73aa2100-3df6-11ea-82ba-47074914e683.png)

However, I know the system is not very reliable. I have noticed that many things can make the process fail (for example, if a user has installed youtube-dl with both **pip** and **apt-get**)

At the moment, I don't understand why the update fails (for some users), and so I don't know how to fix it.

**Comment by axcore** [comment]
> Of course I can update it manually but I wonder why this feature is available while Tartube unable to update youtube-dl by itself?!

I have been some testing. All of the settings work, at least some of the time. 

The correct settings depend on how youtube-dl was installed. Common methods (on Linux Mint) include

Download the source code, and run the binary from the terminal
Download the .DEB, and install it
Install using `sudo apt-get install youtube-dl`
Install using `python3 -m pip install --upgrade youtube-dl`

Everything works on my desktop, but on my laptop, I started getting the same error as you:

`ERROR: Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/usr/bin/youtube-dl'`

I have no idea what the cause of this error is, but the fix is to add `--user`, giving us `python3 -m pip install --upgrade --user youtube-dl`. So I will do that in the next release.

**Comment by axcore** [comment]
Fixed in v1.4.0. If you saw a 'permission denied' error before, you won't see it now.
